data/gwilliams2022.py

```diff
@@ -197,90 +251,161 @@ def shift_brain_signal(self, data, is_Y: bool):
-                        l_freq=self.brain_filter_low,
-                        h_freq=self.brain_filter_high,
-                    )
-
-                    # To 120 Hz
-                    meg_resampled = mne.filter.resample(meg_filtered,
-                                                        down=self.brain_orig_rate /
-                                                        self.brain_resample_rate)  # ( 208, 37853 )
-
-                    # scaler = RobustScaler().fit(meg_resampled)
-                    # meg_scaled = scaler.transform(meg_resampled)
-                    # cprint(meg_scaled.shape, color="cyan")
-
-                    # save to disk
-                    X = np.load(self.x_path, allow_pickle=True).item()
-                    X.update({description: meg_resampled})
-                    np.save(self.x_path, X)
+        consts = dict(
+            num_channels=num_channels,
+            brain_orig_rate=self.brain_orig_rate,
+            brain_filter_low=self.brain_filter_low,
+            brain_filter_high=self.brain_filter_high,
+            brain_resample_rate=self.brain_resample_rate,
+            x_path=self.x_path,
+            root_dir=self.root_dir,
+        )
+
+        subj_list = []
+        for subj in range(num_subjects):
+            for session_idx in range(2):
+                for task_idx in range(4):
+                    subj_list.append((subj, consts, glob_real_durations, session_idx, task_idx))
+
+        # subj_list = [(i, c, rd) for i, c, rd in zip(range(num_subjects), repeat(consts), repeat(real_durations))]
+        with Pool(20) as p:
+            res = list(
+                tqdm(
+                    p.imap(Gwilliams2022Dataset._brain_preproc, subj_list),
+                    total=len(subj_list),
+                    bar_format=bar_format,
+                ))
+        print("no_errors")
+
+        # NOTE: return glob_real_durations (which is a global shared struct) into self.real_durations
+        self.real_durations = dict(glob_real_durations)
+
+        # NOTE: assemble files into one and clean up
+        # FIXME: the paths ending with /0 !!!
+        fnames = [f for f in os.listdir(f"{self.root_dir}/data/Gwilliams2022/preprocessed/0/") if f.startswith("__")]
+        X = dict()
+        for fname in natsorted(fnames):  # NOTE: data MUST be task0, ... taskN, task0, ..., taskN (N=4)
+            X[fname[2:-4]] = np.load(
+                f"{self.root_dir}/data/Gwilliams2022/preprocessed/0/{fname}",
+                allow_pickle=True,
+            )
+        np.save(self.x_path, X)
+        pprint(self.real_durations)
+
+        cprint("removing temp files for EEG data", color="green")
+        for fname in fnames:
+            os.remove(f"{self.root_dir}/data/Gwilliams2022/preprocessed/0/{fname}")
 
         return X
 
     @torch.no_grad()
     def audio_preproc(self):
-        wav2vec = load_wav2vec_model()
+        wav2vec = load_wav2vec_model(self.wav2vec_model)
         wav2vec.eval()
```

Comment:
Rather than calling this in both data processing files, why not code it into `load_wav2vec_model()`?

Reply:
Sure, but I thought (maybe @SeanNobel too) that we can leave this as a parameter, so we could try to push the model and see how it perfoms with a smaller speech model (e.g. `wav2vec2-large`, `wav2vec2-xls`, `wav2vec2-conformer` etc.). There're many smaller models that could just be swapped in for the one we're using.